### PR TITLE
Add a linter for specifications

### PR DESCRIPTION
**Pull request by Elchi3**
As said (and specified) in https://github.com/mdn/stumptown-content/issues/126, here's a linter for the `specifications` front-matter.

### CODE REVIEW

File: scripts/linter/plugins/valid-specs.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                    if (spec === 'non-standard') {
+                      return;
+                    }
+                    // Any other strings should be URLs and provide a deep link to a spec section.
+                    if (!specURLRegex.test(spec)) {
+                        const message = file.message(`"${spec}" is not a valid specification link (anchored deep link required).`, node, ruleId);
+                        message.fatal = true;
+                    }
+                    // Spec URLs should be allow-listed, so that we don't list any outdated specs
+                    let found = false;
```
Comment (ddbeck) [inline scripts/linter/plugins/valid-specs.js]: I found the `let found … if (!found)` sequence a bit hard to follow. What do you think of something like this, instead?

```js
const allowedSpecs = Object.keys(specMap);
if (!allowedSpecs.some(key => spec.includes(key)) {
  const message = file.message(`Domain for "${spec}" not found in data/specifications.yaml`, node, ruleId);
  message.fatal = true;
}
```

File: scripts/linter/plugins/valid-specs.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+const ruleId = "stumptown-linter:valid-specifications";
+const specURLRegex = new RegExp("^http(s)?://[^#]+#.+");
+
+/**
+ * Check the front-matter for valid `specifications`
+ */
+function attacher() {
+    return function transformer(tree, file) {
+        visit(tree, "yaml", node => {
+            let specs = node.data.yaml.specifications;
```
Comment (ddbeck) [inline scripts/linter/plugins/valid-specs.js]:
```suggestion
            const specs = node.data.yaml.specifications;
```

Comment (Elchi3) [reply scripts/linter/plugins/valid-specs.js]: I'm actually modifying `specs` to be always an array in line 24, so I think this should stay `let`.

File: scripts/linter/plugins/valid-specs.js (hunk 1 above)
Comment (Elchi3) [reply scripts/linter/plugins/valid-specs.js]: This is great! Thank you! :)

File: scripts/linter/plugins/valid-specs.js (hunk 2 above)
Comment (ddbeck) [reply scripts/linter/plugins/valid-specs.js]: Oh I missed that `specs` is reassigned below. Disregard!

**Comment by Elchi3** [comment]
Rebased :)

**Comment by ddbeck** [comment]
Oops, that broke against part of the fr language content. Localization surprises

https://travis-ci.org/mdn/stumptown-content/jobs/612402273#L576

**Comment by wbamberg** [comment]
@SphinxKnight , the linter complains that [content/fr/html/HTML.md](https://raw.githubusercontent.com/mdn/stumptown-content/master/content/fr/html/HTML.md) contains:

    specifications: https://html.spec.whatwg.org/multipage/

...where it expects a spec link to contain a fragment. The corresponding file at [content/en-US/html/HTML.md](https://raw.githubusercontent.com/mdn/stumptown-content/master/content/en-US/html/HTML.md) contains:

    specifications: https://html.spec.whatwg.org/multipage/introduction.html#abstract

Is there a reason these should be different?

But also, I wonder if we should relax the check that a spec URL must have a fragment. https://html.spec.whatwg.org/multipage/ seems like a more useful link in this case than https://html.spec.whatwg.org/multipage/introduction.html#abstract.

**Comment by SphinxKnight** [comment]
I'm opening a PR to change the fr version as well :)

**Comment by Elchi3** [comment]
> Oops, that broke against part of the fr language content. Localization surprises

Sorry about that. I should have rebased this again.

**Comment by SphinxKnight** [comment]
*mumble grumpily in French* :)
